crates/pathfinder/src/consensus/inner/consensus_task.rs

```diff
@@ -316,6 +317,14 @@ pub fn spawn(
                                         "🧠 ⏩  {validator_address} catching up current height \
                                          {current_height} -> {cmd_height}",
                                     );
+                                    // TODO: We are initiating catch-up sync but from here onwards
+                                    // the node is still participating in consensus. If we stop
+                                    // before the catch-up is complete, there will be a gap between
+                                    // the blocks that have been synced and `cmd_height` which we
+                                    // won't have a way of filling (as things stand at the moment).
+                                    catch_up_tx
+                                        .send(Some(current_height))
+                                        .expect("Catch-up sync should be running");
```

Comment:
I'm missing a lot of context here, but I though I'd just write what I think about this.

We should be running catch-up sync and consensus concurrently to avoid the issue you're describing. The malachite Starknet node implementation does this too.

In fact, the malachite implementation does the following:

- The sync code is continuously running and feeding synced blocks to consensus (as commit certificates)
- Consensus code is doing the validation and notifies the sync code of the result of the validation (ok, invalid value or other error). The result is used to update peer scoring and is also updating the tip of the chain state for sync.

Reply:
Concurrently running both syncs is something that previously slipped my attention and looks like it's the obvious direction to go :thinking: .